plugins/README.md

```diff
@@ -0,0 +1,88 @@
+## Relayer Plugins
+
+Relayer plugins are TypeScript functions that can be invoked through the relayer HTTP API.
+
+Under the hood, the relayer will execute the plugin code in a separate process using `ts-node` and communicate with it through a Unix socket.
+
+## Setup
+
+#### 1. Writing your plugin
+
+```typescript
+import { Plugin, runPlugin } from "./lib/plugin";
+
+type Args = {
+  foo: string;
+  bar: number;
+}
+
+async function myPlugin(plugin: Plugin, args: Args) {
+    console.log(args.foo);
+    console.log(args.bar);
+
+    const relayer = plugin.useRelayer("my-relayer");
+    await relayer.sendTransaction({
+        to: "0x1234567890123456789012345678901234567890",
+        value: 1000000000000000000,
+    });
+}
+
+runPlugin(myPlugin);
+```
+
+
+#### 2. Adding extra dependencies
+
+You can install any extra JS/TS dependencies in your plugins folder and access them upon execution.
+
+```bash
+npm install ethers
+```
+
+And then just import them in your plugin.
+
+```typescript
+import { ethers } from "ethers";
+```
+
+#### 3. Adding to config file
+
+- id: The id of the plugin. This is used to call a specific plugin through the HTTP API.
+- path: The path to the plugin file - relative to the `/plugins` folder.
+
+```yaml
+{
+  "plugins": [
+    {
+      "id": "my-plugin",
+      "path": "my-plugin.ts"
+    }
+  ]
+}
+```
+
+## Usage
+
+You can call your plugin through the HTTP API, passing your custom arguments as a JSON body.
+
+```bash
+curl -X POST http://localhost:8080/plugins/my-plugin/call -d '{ "params": { "foo": "bar", "bar": 123 } }'
+```
+
+Then the response will include:
+
+- `output`: The output of the plugin execution. Including all the logs from the plugin execution.
```

Comment:
I wonder have you consider using something as result field that would capture value of return statement?
Also, maybe it makes sense to return logs via different field so users could consume it more easier.

Something as:
```
{
  "result": ,   
  "logs": [
    { "level": "info", "message": "...", "timestamp": "..." },
    { "level": "warn", "message": "...", "timestamp": "..." }
  ],
  "traces": [ ... ],         // Raw payloads, as is
  "isError": true,
  "error": {
    "message": "Something went wrong",
    "code": "ERR_CODE",
    "stack": "...",
    "details": { ... }
  }
}
```

Reply:
For autotasks we are monkey patching log commands so we can more easily process logs. maybe that helps here as well.

Reply:
💯 agree

Reply:
I will merge this PR and add the return value + structured output in the next PR